dlhub_sdk/models/servables/python.py

```diff
@@ -163,28 +176,102 @@ def create_model(cls, module, method, autobatch=False, function_kwargs=None):
                 Calls :code:`map(f, list)`
             function_kwargs (dict): Names and values of any other argument of the function to set
                 the values must be JSON serializable.
+            f (object): A function pointer
+            auto_inspect (boolean): Whether or not to attempt to automatically extract inputs from the function
+        Raises:
+            TypeError: If there is no valid way to process the given arguments
         """
+        if f is not None:
+            module, method = f.__module__, f.__name__
+            func = f
+        elif module is not None and method is not None:
+            module_obj = importlib.import_module(module)
+            func = getattr(module_obj, method)
+        else:
+            raise TypeError("PythonStaticMethodModel.create_model was not provided valid arguments. Please provide either a funtion pointer"
+                            " or the module and name of the desired static function")
+
         output = super(PythonStaticMethodModel, cls).create_model(method, function_kwargs)
 
         output.servable.methods["run"].method_details.update({
             'module': module,
             'autobatch': autobatch
         })
+
+        if auto_inspect:
+            sig = Signature.from_callable(func)
+            output = output.set_inputs(**_signature_to_input(sig))
+            output = output.set_outputs(**_signature_to_output(sig))
+
         return output
 
     @classmethod
     def from_function_pointer(cls, f, autobatch=False, function_kwargs=None):
         """Construct the module given a function pointer
-
         Args:
             f (object): A function pointer
             autobatch (bool): Whether to run function on an iterable of entries
             function_kwargs (dict): Any default options for this function
         """
-        return cls.create_model(f.__module__, f.__name__, autobatch, function_kwargs)
+        return cls.create_model(f=f, autobatch=autobatch, function_kwargs=function_kwargs)
 
     def _get_handler(self):
         return 'python.PythonStaticMethodServable'
 
     def _get_type(self):
         return 'Python static method'
+
+
+def _signature_to_input(sig: Signature) -> dict[str, Any]:
+    if len(sig.parameters.values()) == 0:
+        return {"data_type": "python object", "description": "", "python_type": "builtins.NoneType"}  # mirros behavior of LN#276-7
```

Comment:
is this referring to the return of `compose_argument_block()`? I would mention the function name directly, since line numbers change all the time